test/Microsoft.AspNet.Mvc.Core.Test/Internal/ResponseContentTypeHelperTest.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Internal
+{
+    public class ResponseContentTypeHelperTest
+    {
+        public static TheoryData<MediaTypeHeaderValue, string, string> ResponseContentTypeData
+        {
+            get
+            {
+                // contentType, responseContentType, expectedContentType
+                return new TheoryData<MediaTypeHeaderValue, string, string>
+                {
+                    {
+                        null,
+                        null,
+                        "text/default; p1=p1-value; charset=utf-8"
+                    },
+                    {
+                        new MediaTypeHeaderValue("text/foo"),
+                        null,
+                        "text/foo"
+                    },
+                    {
+                        MediaTypeHeaderValue.Parse("text/foo; charset=us-ascii"),
+                        null,
+                        "text/foo; charset=us-ascii"
+                    },
+                    {
+                        MediaTypeHeaderValue.Parse("text/foo; p1=p1-value"),
+                        null,
+                        "text/foo; p1=p1-value"
+                    },
+                    {
+                        MediaTypeHeaderValue.Parse("text/foo; p1=p1-value; charset=us-ascii"),
+                        null,
+                        "text/foo; p1=p1-value; charset=us-ascii"
+                    },
+                    {
+                        null,
+                        "text/bar",
+                        "text/bar"
+                    },
+                    {
+                        null,
+                        "text/bar; p1=p1-value",
+                        "text/bar; p1=p1-value"
+                    },
+                                        {
+                        null,
+                        "text/bar; p1=p1-value; charset=us-ascii",
+                        "text/bar; p1=p1-value; charset=us-ascii"
+                    },
+                    {
+                        MediaTypeHeaderValue.Parse("text/foo; charset=us-ascii"),
+                        "text/bar",
+                        "text/foo; charset=us-ascii"
+                    },
+                    {
+                        MediaTypeHeaderValue.Parse("text/foo; charset=us-ascii"),
+                        "text/bar; charset=utf-8",
+                        "text/foo; charset=us-ascii"
+                    }
```

Comment:
It would be nice to document user-scenarios each of these test inputs represent. I don't feel like using `[Theory]` here makes these tests better, it's hard to tell what's tested and what's not.